src/Plugin.php

```diff
@@ -126,6 +155,22 @@ public function handleEvent(Event $event, Queue $queue)
         $event->getConnection()->setNickname($nick);
     }
 
+    /**
+     * Handle primary nick recovery.
+     *
+     * @param \Phergie\Irc\Event\UserEventInterface $event
+     * @param \Phergie\Irc\Bot\React\EventQueueInterface $queue
+     */
+    public function handleQuit(UserEvent $event, Queue $queue)
+    {
+        $nick = $event->getNick();
+        if ($this->primaryNick !== null && $nick == $this->primaryNick) {
+            $this->logger->debug("[AltNick] '$nick' disconnected, switching to primary nick");
+            $queue->ircNick($this->primaryNick);
```

Comment:
Unless it's in core client now AND we bump up dependencies to require a version with that functionality, we should explicitly change connection nick.

If / when both of those conditions are true, we can update multiple location in this plugin to remove those explicit calls to change nick in connection.